lib/npg_pipeline/pluggable.pm

```diff
@@ -612,6 +642,25 @@ sub _run_function {
     $attrs->{$key} = $value;
   }
 
+  ####
+  # Bring in resource requirements that functions need in order to
+  # create task definitions
+
+  # Extract default properties from graphwide metadata
+  my $jgraph = $self->_function_list_conf;
+  my $resource = $jgraph->{graph}{metadata}{default_resources};
```

Comment:
Yes. I would prefer to do it with a schema, perhaps evaluated at every pipeline launch rather than in a unit test. Need approval to put the time into the schema though.